test/unit/org/apache/cassandra/repair/ConcurrentIrWithPreviewFuzzTest.java

```diff
@@ -62,7 +64,7 @@ public void concurrentIrWithPreview()
                 // cause a delay in validation to have more failing previews
                 closeables.add(cluster.nodes.get(pickParticipant(rs, previewCoordinator, preview)).doValidation(next -> (cfs, validator) -> {
                     if (validator.desc.parentSessionId.equals(preview.state.id))
-                        cluster.unorderedScheduled.schedule(() -> next.accept(cfs, validator), 1, TimeUnit.HOURS);
+                        delayValidation(cluster, ir, next, cfs, validator);
```

Comment:
this was a flakey issue that this patch discovered... with IR retries we can have IR take longer than 1h, so waiting by time is unsafe and leads to flakey builds... now we check IR state and keep retrying until IR completes